server/cli/src/test_connection.rs

```diff
@@ -0,0 +1,499 @@
+extern crate machine_uid;
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use clap::Parser;
+use log::*;
+use repository::{get_storage_connection_manager, migrations::Version};
+use reqwest::{Client, Url};
+use serde::{Deserialize, Serialize};
+use server::configuration;
+use service::{
+    apis::login_v4::{LoginApiV4, LoginInputV4, LoginUserTypeV4},
+    app_data::{AppDataService, AppDataServiceTrait},
+    service_provider::ServiceProvider,
+    settings::is_develop,
+    sync::{
+        api::{SyncApiSettings, SyncApiV5},
+        api_v6::SyncApiV6,
+        settings::SyncSettings,
+    },
+};
+use tokio::sync::mpsc;
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[arg(short, long)]
+    gui: bool,
+    #[arg(short, long)]
+    username: Option<String>,
+    #[arg(short, long)]
+    password: Option<String>,
+    #[arg(short, long)]
+    server_uuid: Option<String>,
+}
+
+#[tokio::main]
+async fn main() {
+    simple_logger::init_with_level(Level::Info).unwrap();
+
+    let args = Args::parse();
+    let gui = args.gui;
+
+    let (gui_tx, gui_rc) = mpsc::channel(10);
+    let test_task = tokio::spawn(perform_tests(gui_tx, args));
+
+    if gui {
+        let options = eframe::NativeOptions {
+            viewport: egui::ViewportBuilder::default().with_inner_size([320.0, 240.0]),
```

Comment:
Not a blocker, but if we're making it this small, is there a way to add a scrollbar? 😁